src/main/java/com/duolingo/open/rtlviewpager/DelegatingPagerAdapter.java

```diff
@@ -18,75 +18,52 @@
 
 import android.database.DataSetObserver;
 import android.os.Parcelable;
-import android.support.annotation.NonNull;
-import android.support.v4.view.PagerAdapter;
+import androidx.annotation.NonNull;
+import androidx.viewpager.widget.PagerAdapter;
 import android.view.View;
 import android.view.ViewGroup;
 
 public class DelegatingPagerAdapter extends PagerAdapter {
+
     @NonNull
     private final PagerAdapter mDelegate;
 
-    public DelegatingPagerAdapter(@NonNull final PagerAdapter delegate) {
+    DelegatingPagerAdapter(@NonNull final PagerAdapter delegate) {
         this.mDelegate = delegate;
         delegate.registerDataSetObserver(new MyDataSetObserver(this));
     }
 
     @NonNull
-    public PagerAdapter getDelegate() {
+    PagerAdapter getDelegate() {
         return mDelegate;
     }
 
     public int getCount() {
         return mDelegate.getCount();
     }
 
-    public void startUpdate(ViewGroup container) {
+    public void startUpdate(@NonNull ViewGroup container) {
         mDelegate.startUpdate(container);
     }
 
-    public Object instantiateItem(ViewGroup container, int position) {
-        return mDelegate.instantiateItem(container, position);
-    }
-
-    public void destroyItem(ViewGroup container, int position, Object object) {
-        mDelegate.destroyItem(container, position, object);
-    }
-
-    public void setPrimaryItem(ViewGroup container, int position, Object object) {
-        mDelegate.setPrimaryItem(container, position, object);
-    }
-
-    public void finishUpdate(ViewGroup container) {
-        mDelegate.finishUpdate(container);
-    }
-
-    @Deprecated
```

Comment:
We should leave the deprecated methods since they are still part of the public API for the superclass.

Reply:
Ok, done